lecli/log_api.py

```diff
@@ -0,0 +1,174 @@
+
+
+def get_logs():
+    """
+    Get logs associated with the user.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.request('GET', _url(), headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def get_log(log_id):
+    """
+    Get a specific log.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.get(_url() + "/" + str(log_id),
+                                headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def create_log(logname, params):
+    """
+    Add a new log to the current account.
+    If a filename is given, load the contents of the file
+    as json parameters for the request.
+    If a name is given, create a new log with the given name
+    """
+
+    if params is not None:
+        request_params = params
+    else:
+        request_params = {
+            'log': {
+                'name': str(logname)
+            }
+        }
+
+    headers = api_utils.generate_headers('rw')
+
+    try:
+        response = requests.post(_url(), json=request_params, headers=headers)
+        if response_utils.response_error(response):
+            print 'Creating log failed, status code: %d' % response.status_code
+            exit(1)
+        elif response.status_code == 201:
+            pretty_print_string_as_json(response.text)
+
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def delete_log(log_id):
+    """
+    Delete a log with the provided log ID.
+    """
+    url = _url() + '/' + log_id
+    headers = api_utils.generate_headers('rw')
+
+    try:
+        response = requests.delete(url, headers=headers)
+        if response_utils.response_error(response):
+            print 'Delete log failed, status code: %d' % response.status_code
+            exit(1)
+        elif response.status_code == 204:
+            print 'Deleted log with id: %s' % log_id
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def patch_log(log_id, url, headers, params):
+    """Makes a HTTP PATCH call with the given details"""
+    try:
+        response = requests.patch(url, json=params, headers=headers)
+        if response_utils.response_error(response):
+            print 'Updating log with details: %s failed, status code: %d' \
+                  % (params, response.status_code)
+            exit(1)
+        elif response.status_code == 200:
+            print "Log: '%s' updated to: %s"  % (log_id, params)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def rename_log(log_id, log_name):
+    """
+    Update the given log with the name provided
+    """
+    url = _url() + '/' +str(log_id)
```

Comment:
i think os.path.join is for OS file system paths, for web URLs something like `"/".join([_url(), log_id])` sounds like the right call.